doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -11,92 +11,83 @@ These tactics use the equality :g:`eq:forall A:Type, A->A->Prop` defined in
 file ``Logic.v`` (see :ref:`coq-library-logic`). The notation for :g:`eq T t u` is
 simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
 
-.. tacn:: rewrite @term
-   :name: rewrite
+.. tacn:: rewrite {+, @oriented_rewriter } {? @occurrences } {? by @ltac_expr3 }
 
-   This tactic applies to any goal. The type of :token:`term` must have the form
+   .. insertprodn oriented_rewriter constr_with_bindings_arg
 
-   ``forall (x``:sub:`1` ``:A``:sub:`1` ``) ... (x``:sub:`n` ``:A``:sub:`n` ``), eq term``:sub:`1` ``term``:sub:`2` ``.``
+   .. prodn::
+      oriented_rewriter ::= {? {| -> | <- } } {? @natural } {? {| ? | ! } } @constr_with_bindings_arg
+      constr_with_bindings_arg ::= {? > } @one_term {? with @bindings }
 
-   where :g:`eq` is the Leibniz equality or a registered setoid equality.
+   Rewrites terms based on the equality given by :n:`@one_term`, which
+   must have the form:
 
-   Then :n:`rewrite @term` finds the first subterm matching `term`\ :sub:`1` in the goal,
-   resulting in instances `term`:sub:`1`' and `term`:sub:`2`' and then
-   replaces every occurrence of `term`:subscript:`1`' by `term`:subscript:`2`'.
-   Hence, some of the variables :g:`x`\ :sub:`i` are solved by unification,
-   and some of the types :g:`A`\ :sub:`1`:g:`, ..., A`\ :sub:`n` become new
-   subgoals.
+   :n:`forall (x__1: A__1) ... (x__n: A__n), eq term__1 term__2`
 
-   .. exn:: The @term provided does not end with an equation.
-      :undocumented:
-
-   .. exn:: Tactic generated a subgoal identical to the original goal. This happens if @term does not occur in the goal.
-      :undocumented:
-
-   .. tacv:: rewrite -> @term
-
-      Is equivalent to :n:`rewrite @term`
+   where :g:`eq` is the Leibniz equality or a registered setoid equality.
```

Comment:
We should say that, in the latter case, a different implementation is used, and link to the "Generalized rewriting" chapter for more about setoid rewriting.

Reply:
comme ça?

```
   where :g:`eq` is the Leibniz equality.  Use :tacn:`setoid_rewrite`, described in
   :ref:`generalizedrewriting`, when `eq` is a setoid equality.
```